src/core/services/interimElement/interimElementBase.js

```diff
@@ -0,0 +1,217 @@
+          hideElement().then(function() {
+
+            (isCancelled && API.deferred.reject(response)) || API.deferred.resolve(response);
+
+          },  API.deferred.reject);
+        });
+
+        return API.deferred.promise;
+      }
+    }
+
+    function createAndShowElement() {
+      var defer = $q.defer();
+
+      compileElement(options).then(function(compiledData) {
+        element = linkElement(compiledData);
+
+        showAction = showElement(compiledData.controller).then(defer.resolve, rejectAll);
+      }, rejectAll);
+
+      function rejectAll(error) {
+        API.deferred.reject(error);
+        defer.reject(error);
+      }
+
+      return defer.promise;
+    }
+
+    function showElement(controller) {
+      var notifyShowing = options.onShowing || angular.noop;
+      var notifyComplete = options.onComplete || angular.noop;
+
+      // Notify the `onShowing` function before the show process starts.
+      notifyShowing(options.scope, element, options, controller);
+
+      var defer = $q.defer();
+
+      try {
+        $q.when(options.onShow(options.scope, element, options, controller))
+          .then(function() {
+
+            notifyComplete(options.scope, element, options);
+            startAutoHide();
+
+            defer.resolve(element);
+
+          }, defer.reject)
+
+      } catch (err) {
+        defer.reject(err.message);
+      }
+
+      return defer.promise;
+    }
+
+    function hideElement() {
+      var defer = $$q.defer();
+      var announceRemoving = options.onRemoving || angular.noop;
+
+      try {
+        var action = $$q.when(options.onRemove(options.scope, element, options) || true);
+
+        announceRemoving(element, action);
+
+        if (options.$destroy) {
+          defer.resolve(element);
+        } else {
+
+          action.then(function() {
+
+            if (!options.preserveScope && options.scope) {
+              options.scope.$destroy();
+            }
+
+            defer.resolve(element);
+
+          }, defer.reject);
+
+        }
+
+      } catch (err) {
+        defer.reject(err.message);
+      }
+
+      return defer.promise;
+    }
+
+    function prepareInterimElement() {
+      if (options.template) {
+        options.template = $mdUtil.processTemplate(options.template);
+      }
+
+      // Apply default options to the interim element.
+      options = angular.extend({
+        preserveScope: false,
+        cancelAutoHide : angular.noop,
+        scope: options.scope || $rootScope.$new(options.isolateScope),
+
+        onShow: function transitionIn(scope, element, options) {
+          return $animate.enter(element, options.parent);
+        },
+
+        onRemove: function transitionOut(scope, element) {
+          return element && $animate.leave(element) || $q.when();
+        }
+      }, options);
+    }
+
+
+    function compileElement(options) {
+      return options.skipCompile ? mockCompile() : $mdCompiler.compile(options);
```

Comment:
Yes I agree, that's possible, I just thought, that it would be clearer without inlining.